src/clzones.h

```diff
@@ -234,6 +235,10 @@ class zone_manager
         std::unordered_set<tripoint> get_point_set( const zone_type_id &type ) const;
         std::unordered_set<tripoint> get_vzone_set( const zone_type_id &type ) const;
 
+        //Cache to prevent rechecking of items when sorting
+        std::unordered_map<tripoint, std::queue<item *> *> to_sort;
```

Comment:
Consider making this a std::queue<item *, std::vector>.  The default queue container (std::deque) has very inconsistent and poorly-behaving implementations across various platforms.